src/macros.rs

```diff
@@ -327,3 +327,148 @@ macro_rules! template {
     };
     () => {}
 }
+
+/// Utility macro for conditionally joining labels, such as `String` or `&str`,
```

Comment:
That's a bit ambiguous. The labels are *unconditionally* joined but *conditionally* included. "Utility macro for joining a set of conditionally included labels, ..." is the best I could come up with but that still sounds a bit awkward.